tests/test_mirror_port_erspan.py

```diff
@@ -0,0 +1,550 @@
+        self.dvs_mirror.verify_session_status(session, status="inactive")
+
+        # create neighbor to vlan 6
+        dvs.add_neighbor(vlan, "6.6.6.6", "66:66:66:66:66:66")
+        self.dvs_mirror.verify_session_status(session, status="inactive")
+
+        # create fdb entry to ethernet4
+        dvs.create_fdb(vlan_id, "66-66-66-66-66-66", "Ethernet4")
+        self.dvs_mirror.verify_session_status(session)
+
+        src_mac = dvs.runcmd("bash -c \"ip link show eth0 | grep ether | awk '{print $2}'\"")[1].strip().upper()
+        expected_asic_db = {"SAI_MIRROR_SESSION_ATTR_MONITOR_PORT": pmap.get("Ethernet4"),
+                            "SAI_MIRROR_SESSION_ATTR_TYPE": "SAI_MIRROR_SESSION_TYPE_ENHANCED_REMOTE",
+                            "SAI_MIRROR_SESSION_ATTR_ERSPAN_ENCAPSULATION_TYPE": "SAI_ERSPAN_ENCAPSULATION_TYPE_MIRROR_L3_GRE_TUNNEL",
+                            "SAI_MIRROR_SESSION_ATTR_IPHDR_VERSION": "4",
+                            "SAI_MIRROR_SESSION_ATTR_TOS": "32",
+                            "SAI_MIRROR_SESSION_ATTR_TTL": "100",
+                            "SAI_MIRROR_SESSION_ATTR_SRC_IP_ADDRESS": "5.5.5.5",
+                            "SAI_MIRROR_SESSION_ATTR_DST_IP_ADDRESS": "6.6.6.6",
+                            "SAI_MIRROR_SESSION_ATTR_DST_MAC_ADDRESS": "66:66:66:66:66:66",
+                            "SAI_MIRROR_SESSION_ATTR_SRC_MAC_ADDRESS": src_mac,
+                            "SAI_MIRROR_SESSION_ATTR_GRE_PROTOCOL_TYPE": "25944",
+                            "SAI_MIRROR_SESSION_ATTR_VLAN_HEADER_VALID": "true",
+                            "SAI_MIRROR_SESSION_ATTR_VLAN_TPID": "33024",
+                            "SAI_MIRROR_SESSION_ATTR_VLAN_ID": vlan_id,
+                            "SAI_MIRROR_SESSION_ATTR_VLAN_PRI": "0",
+                            "SAI_MIRROR_SESSION_ATTR_VLAN_CFI": "0"}
+        self.dvs_mirror.verify_session(dvs, session, asic_db=expected_asic_db, src_ports=src_asic_ports, asic_size=16, direction="TX")
+
+        dvs.set_interface_status("Ethernet4", "down")
+
+        # remove fdb entry
+        dvs.remove_fdb(vlan_id, "66-66-66-66-66-66")
+        self.dvs_mirror.verify_session_status(session, status="inactive")
+
+        # remove neighbor
+        dvs.remove_neighbor(vlan, "6.6.6.6")
+        self.dvs_mirror.verify_session_status(session, status="inactive")
+
+        # remove ip address
+        dvs.remove_ip_address(vlan, "6.6.6.0/24")
+        self.dvs_mirror.verify_session_status(session, status="inactive")
+
+        # bring down vlan and member
+        dvs.set_interface_status("Ethernet4", "down")
+        dvs.set_interface_status(vlan, "down")
+
+        # remove mirror session
+        self.dvs_mirror.remove_mirror_session(session)
+        self.dvs_mirror.verify_no_mirror()
+
+        # remove vlan member; remove vlan
+        self.dvs_vlan.remove_vlan_member(vlan_id, "Ethernet4")
+        self.dvs_vlan.remove_vlan(vlan_id)
```

Comment:
In general I recommend checking that there's no vlan/no vlan member at the end of the test case just to ensure that the test case leaves the system in a clean state.

Reply:
Done. Added checks.